--- a/com/mojang/minecraft/client/renderer/DirtyChunkSorter.java
+++ b/com/mojang/minecraft/client/renderer/DirtyChunkSorter.java
@@ -14,10 +14,9 @@
     // $FF: bridge method
     public final int compare(Object objOne, Object objTwo) {
         RenderingChunk renderingchunk = (RenderingChunk) objOne;
-        objTwo = (RenderingChunk) objTwo;
-        objOne = renderingchunk;
-        boolean flag = objOne.visible;
-        boolean flag1 = objTwo.visible;
-        return flag && !flag1 ? -1 : ((!flag1 || flag) && objOne.compare(this.player) < objTwo.compare(this.player) ? -1 : 1);
+        RenderingChunk renderingchunk1 = (RenderingChunk) objTwo;
+        boolean flag = renderingchunk.visible;
+        boolean flag1 = renderingchunk1.visible;
+        return flag && !flag1 ? -1 : ((!flag1 || flag) && renderingchunk.compare(this.player) < renderingchunk1.compare(this.player) ? -1 : 1);
     }
 }
